src/PhpParser/Node/NamedVariableFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\PhpParser\Node;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Variable;
+use Rector\Core\Exception\ShouldNotHappenException;
+use Rector\Naming\Naming\VariableNaming;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+final class NamedVariableFactory
+{
+    public function __construct(
+        private VariableNaming $variableNaming
+    ) {
+    }
+
+    public function createVariable(\PhpParser\Node $node, string $variableName): Variable
+    {
+        $currentStmt = $node->getAttribute(AttributeKey::CURRENT_STATEMENT);
```

Comment:
should can be replaced by `BetterNodeFinder->resolveCurrentStatement()` with set method modifier to public

https://github.com/rectorphp/rector-src/blob/80a04aa51f57c801cf30b81b3c02f3480471f278/src/PhpParser/Node/BetterNodeFinder.php#L541

Reply:
Event better, thanks :+1: